pyega3/libs/data_file.py

```diff
@@ -38,8 +38,15 @@ def __init__(self, data_client, file_id,
         self._unencrypted_checksum = unencrypted_checksum
         self._file_status = status
 
+    @staticmethod
+    def from_metadata(data_client, metadata):
+        file_id = metadata['fileId']
+        result = DataFile(data_client, file_id)
+        result._set_metadata_from_json(metadata)
+        return result
+
     def load_metadata(self):
-        res = self.data_client.get_json(f"/metadata/files/{self.id}")
+        res = self.data_client.get_json(f"/files/{self.id}")
 
         # If the user does not have access to the file then the server returns HTTP code 200 but the JSON payload has
         # all the fields empty
```

Comment:
Please check the line after this one is using `unencryptedChecksum` but should be renamed to `plainChecksum`

Reply:
I changed it so that it only does the check in the v1 API, I think in the V2 API we should never return empty metadata objects like this, we should return proper error codes instead :)